src/main/java/teammates/ui/webapi/action/GetSessionResultsAsCsvAction.java

```diff
@@ -0,0 +1,84 @@
+package teammates.ui.webapi.action;
+
+import teammates.common.datatransfer.SectionDetail;
+import teammates.common.datatransfer.attributes.FeedbackSessionAttributes;
+import teammates.common.datatransfer.attributes.InstructorAttributes;
+import teammates.common.exception.EntityDoesNotExistException;
+import teammates.common.exception.EntityNotFoundException;
+import teammates.common.exception.ExceedingRangeException;
+import teammates.common.exception.InvalidHttpParameterException;
+import teammates.common.exception.RequestExceedingRangeException;
+import teammates.common.util.Const;
+
+/**
+ * Gets feedback session results in csv.
+ */
+public class GetSessionResultsAsCsvAction extends Action {
+
+    @Override
+    protected AuthType getMinAuthLevel() {
+        return AuthType.PUBLIC;
+    }
+
+    @Override
+    public void checkSpecificAccessControl() {
+        String courseId = getNonNullRequestParamValue(Const.ParamsNames.COURSE_ID);
+        String feedbackSessionName = getNonNullRequestParamValue(Const.ParamsNames.FEEDBACK_SESSION_NAME);
+
+        FeedbackSessionAttributes fs = logic.getFeedbackSession(feedbackSessionName, courseId);
+
+        if (fs == null) {
+            throw new EntityNotFoundException(new EntityDoesNotExistException("Feedback session is not found"));
+        }
+
+        InstructorAttributes instructor = logic.getInstructorForGoogleId(courseId, userInfo.id);
+        gateKeeper.verifyAccessible(instructor, fs);
+    }
+
+    @Override
+    public ActionResult execute() {
+        String section = getRequestParamValue(Const.ParamsNames.SECTION_NAME);
+        String sectionDetailValue = getRequestParamValue(Const.ParamsNames.SECTION_NAME_DETAIL);
+        SectionDetail sectionDetail = SectionDetail.NOT_APPLICABLE;
+        if (section != null && sectionDetailValue != null && !sectionDetailValue.isEmpty()) {
+            if (SectionDetail.containsSectionDetail(sectionDetailValue) == null) {
```

Comment:
I honestly feel `SectionDetail.containsSectionDetail` should return true/false instead. You are not even using the value it returns here.

Reply:
Yes I think you are right. I'm not sure why `SectionDetail.containsSectionDetail` returns `SectionDetail` object since the value does not seem to be used anywhere else either. I will change the return type of that function.